app/Http/Controllers/TradeOfferController.php

```
@@ -0,0 +1,122 @@
+    public function create(Request $request)
+    {
+        //REQUEST DOES NOT HAVE TRADE_ID NOR CHARACTER_ID . VALIDATION FAILS
+        $offerData = $request->validate([
+            'trade_id' => 'required|exists:trades,id|integer',
+            'character_id' => "required|exists:characters,id|integer",
+        ]);
+
+        $character = Character::where('user_id', Auth::id())
+            ->findOrFail($offerData['character_id']);
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $tradeItem = $trade->item()->get();
+
+        $validItems = $character->items()->where('tier', $tradeItem->tier);
+
+        return view('offer.create', compact('validItems'));
+    }
+
+    /**
+     * Retrieve Trade Offer to edit
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Request $request)
+    {
+        //rrequest params passed in "headers" instead of request parameters...
+
+        $offerData = $request->validate([
+            'trade_id' => 'required|exists:trades,id|integer',
+            'item_id' => 'required|exists:items,id|integer',
+        ]);
+
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $item = Item::findOrFail($offerData['item_id']);
+
+        return view('offer.edit', compact('item', 'trade'));
+    }
+
+    /**
+     *store trade offer
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(Request $request)
+    {
+        $offerData = $request->validate([
+            'item_id' => 'required|exists:items,id|integer',
+            'trade_id' => 'required|exists:trades,id|integer'
+        ]);
+
+        //findOrFail redundant?
+        $trade = Trade::findOrFail($offerData['trade_id']);
+        $dataItem = Item::findOrFail($offerData['item_id']);
+        $offerItem = ($dataItem->tier == $trade->item()->get()[0]->tier) ? $dataItem : null;
```

if the offered item is the wrong rarity you should actually raise an exception or return back with an error of some sort.